libs/vscode/messaging/src/lib/pipe.ts

```diff
@@ -0,0 +1,49 @@
+import { mkdirSync, unlinkSync } from 'fs';
+import { platform, tmpdir } from 'os';
+import { join, resolve } from 'path';
+import { importNxPackagePath } from '@nx-console/shared-npm';
+
+const DAEMON_DIR_FOR_CURRENT_WORKSPACE = join('.nx', 'workspace-data', 'd');
+
+async function socketDirName(workspaceRoot: string): Promise<string> {
+  const { hashArray } = await importNxPackagePath<
+    typeof import('nx/src/native')
+  >(workspaceRoot, 'src/native');
+  const unique = hashArray([workspaceRoot, 'nx-console']);
+  return join(tmpdir(), unique);
+}
+
+async function getSocketDir(workspaceRoot: string) {
+  try {
+    const dir =
+      process.env.NX_SOCKET_DIR ??
+      process.env.NX_DAEMON_SOCKET_DIR ??
+      (await socketDirName(workspaceRoot));
+    if (platform() !== 'win32') {
+      mkdirSync(dir, { recursive: true });
+    }
+    return dir;
+  } catch (e) {
+    return join(workspaceRoot, DAEMON_DIR_FOR_CURRENT_WORKSPACE);
+  }
+}
+
+export const getFullOsSocketPath = async (workspaceRoot: string) => {
```

Comment:
should we allow ppl to modify this somehow if they have too long socket paths? Like we do with the daemon in the nx repo?

Reply:
They can use the same env variables that Nx uses. And I don't think we'll ever hit that max path with the hash from Nx. The max amount of characters that produces is going to be the length of a 64 bit integer (which the max number would be `18446744073709551615`) so the max length would be 20 characters here + `nx-console.sock`